go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -297,6 +299,13 @@ func (oc *DefaultNetworkController) syncAddressSetsAndAcls() error {
 		return fmt.Errorf("failed to sync acls on controller init: %v", err)
 	}
 
+	// port groups should be synced only once across all controllers.
+	// Do it here since DefaultNetworkController is always created, and this sync has dependencies with the other syncs
+	// in this function
```

Comment:
added a comment

Reply:
yeah I agree it has a mixed purpose, but i couldn't figure out a better place (not a fan of adding more utils), maybe you have some bright ideas where it would make more sense?

Reply:
NetworkControllerManager comes to mind?

Reply:
I tried to move some stuff, but still don't like it :D
I don't think DefaultNetworkControllerName should be used outside the ovn package (and it looks like it's not!)
the dependency is that default network controller should update its db first, then all secondary networks should update port groups, and the cleanup part is a regular cleanup and not a migration. Whatever I do with this function it only gets worse ><

Reply:
It's just moving the whole thing "as is" to another higher level place, like, I am not suggesting to split it by networks; or I am missing something else?
DefaultNetworkControllerName looks like something that could be globally defined. Isn't it used in the node package as well?
Anyway, we can move on I guess. It will not be intuitive to find where secondary network stuff is migrated if it is in the default network controller, but I guess that is not the worse thing in the world.